pgml-dashboard/src/utils/url_parser.rs

```diff
@@ -0,0 +1,128 @@
+use lazy_static::lazy_static;
+use regex::Regex;
+
+lazy_static! {
+    static ref DOMAIN_REGEX: Regex =
+        Regex::new(r"^https?://((localhost:8000)|(.*\.?postgresml\.org))").unwrap();
+    static ref UUID_REGEX: Regex = Regex::new(".{8}-.{4}-.{4}-.{4}-.{12}").unwrap();
+}
+
+pub struct ClusterUrlParser;
+
+impl ClusterUrlParser {
+    // Returns option containing path. If regex does not match, returns None.
+    pub fn get_path(url: &str) -> Option<String> {
+        let path = DOMAIN_REGEX.replace(url.clone(), "").to_string();
+        match url.eq(&path) {
+            true => None,
+            _ => Some(path),
+        }
+    }
+
+    // Finds the uuid in a url.
+    pub fn get_uuid(uri: String) -> Option<String> {
+        let uuid = UUID_REGEX.find(&uri);
+        match uuid {
+            Some(uuid) => Some(uri.to_string()[uuid.start()..uuid.end()].to_string()),
+            None => None,
+        }
+    }
+
+    // Determines if there is a uuid in a url.
+    pub fn has_uuid(uri: String) -> bool {
+        match UUID_REGEX.find(&uri) {
+            Some(_) => true,
+            _ => false,
+        }
+    }
+
+    // Replaces the first uuid with a new uuid in a url.
+    pub fn replace_path_uuid(path: String, uuid: String) -> String {
```

Comment:
I don't see these uuid methods used anywhere.

Reply:
This is used in src/api/cluster/mod line 223 of hypercloud repo.  In fact, since a user can only have one local db, we can move the entire utility to the other repo.